examples/99-advanced/customization-trame-toolbar.py

```diff
@@ -0,0 +1,155 @@
+# and flush the server, i.e. "force" the server to see the change.
+# We will see more on the state variables in a bit, but we need to create the
+# function here otherwise the server will complain ``btn_play`` does not exist.
+
+
+def btn_play():
+    state.play = not state.play
+    state.flush()
+
+
+###############################################################################
+# We will do a simple rendering of a Cone using the vtk `vtkConeSouce`
+# algorithm.
+#
+# When using the ``pl.show`` method. The function we created ``custom_tools``
+# should be passed as a ``jupyter_kwargs`` argument under the key
+# ``add_menu_items``.
+
+pl = pv.Plotter(notebook=True)
+algo = vtk.vtkConeSource()
+mesh_actor = pl.add_mesh(algo)
+
+widget = pl.show(jupyter_kwargs=dict(add_menu_items=custom_tools), return_viewer=True)
+
+###############################################################################
+# To interact with ``trame``'s server we need to get the server's state.
+#
+# We initialize the ``play`` variable in the shared state and this will be
+# controlled by the play button we created. Note that when creating the
+# ``slider``, the ``text_field`` and the ``select`` tools, we passed something
+# like ``model=("variable", value). This will automatically create the variable
+# "variable" with value ``value`` in the server's shared state, so we do not need
+# to create ``state.resolution`` or ``state.visibility``.
+
+state, ctrl = widget.viewer.server.state, widget.viewer.server.controller
+state.play = False
+ctrl.view_update = widget.viewer.update
+
+###############################################################################
+# Now we can create the callback functions for our menu items.
+#
+# The functions are decorated with a ``state.change("variable")``. This means
+# they will be called when this specific variable has its value changed in the
+# server's shared state. When ``resolution`` changes, we want to update the
+# resolution of our cone algorithm. When ``visibility`` changes, we want to toggle the
+# visibility of our cone.
+#
+# The ``play`` variable is a little bit trickier. We want to start something like
+# a timer so that an animation can be set to play. To do that with ``trame`` we need
+# to have an asynchronous function so we can continue to do stuff while the
+# "timer" function is running. The ``_play`` function will be called when the ``play``
+# variable is changed (when we click the play button, through the ``btn_play``
+# callback). While ``state.play`` is ``True`` we want to play the animation. We
+# change the ``state.resolution`` value, but to really call the ``update_resolution``
+# function we need to ``flush`` the server and force it to see the change in
+# the shared variables. When ``state.play`` changes to ``False``, the animation stops.
+#
+# Note that using ``while play: ...`` would not work here because it is not the
+# actual state variable, but only an argument value passed to the callback function.
```

Comment:
```suggestion
# Now we can create the callback functions for our menu items.
#
# The functions are decorated with a ``state.change("variable")``. This means
# they will be called when this specific variable has its value changed in the
# server's shared state. When ``resolution`` changes, we want to update the
# resolution of our cone algorithm. When ``visibility`` changes, we want to toggle the
# visibility of our cone.
#
# The ``play`` variable is a little bit trickier. We want to start something like
# a timer so that an animation can be set to play. To do that with ``trame`` we need
# to have an asynchronous function so we can continue to do stuff while the
# "timer" function is running. The ``_play`` function will be called when the ``play``
# variable is changed (when we click the play button, through the ``button_play``
# callback). While ``state.play`` is ``True`` we want to play the animation. We
# change the ``state.resolution`` value, but to really call the ``update_resolution``
# function we need to ``flush`` the server and force it to see the change in
# the shared variables. When ``state.play`` changes to ``False``, the animation stops.
#
# Note that using ``while play: ...`` would not work here because it is not the
# actual state variable, but only an argument value passed to the callback function.
```